spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -168,6 +169,9 @@ protected void onInit() {
 					this.taskExecutor = new ErrorHandlingTaskExecutor(this.taskExecutor, this.errorHandler);
 				}
 			}
+			if (this.transactionSynchronizationFactory == null) {
+				this.transactionSynchronizationFactory = new PassthroughTransactionSynchronizationFactory();
+			}
```

Comment:
OK, Gary! 

That is fully different story. Sorry, that wasn't obvious by your comment that you wouldn't like to hit `ThreadLocal` 😄 

Does that mean that we have to revise all our `TSM.iATA()` usage
For example `ConnectionFactoryUtils`:
```
public static RabbitResourceHolder bindResourceToTransaction(RabbitResourceHolder resourceHolder,
		ConnectionFactory connectionFactory, boolean synched) {
	if (TransactionSynchronizationManager.hasResource(connectionFactory)
			|| !TransactionSynchronizationManager.isActualTransactionActive() || !synched) {
		return (RabbitResourceHolder) TransactionSynchronizationManager.getResource(connectionFactory);
	}
```
And `ImapIdleChannelAdapter`:
```
if (TransactionSynchronizationManager.isActualTransactionActive()) {
		if (ImapIdleChannelAdapter.this.transactionSynchronizationFactory != null) {
```
?

I don't mind about optimization. 
So, another trick is check if our `adviceChain` contains `TransactionInterceptor`.
If it is there and `TSF == null` we populate this our new `PassthroughTransactionSynchronizationFactory`

Does that work for you?